src/score_hv/stats_utils.py

```diff
@@ -0,0 +1,38 @@
+"""Copyright 2024 NOAA
+All rights reserved.
+
+Methods to calculate gridcell area weighted statistics
+The functions generally take the following two input variables:
+    xarray_variable - xarray variable data
+    gridcell_area_data - xarray variable containing gridcell areas, in the
+    same shape as xarray_variable
+"""
+
+import numpy as np
+
+def area_weighted_mean(xarray_variable, gridcell_area_weights):  
+    """returns the gridcell area weighted mean of the xarray_variable
+    """
+    weighted_mean, sumweights = np.ma.average(xarray_variable,
+                                              weights=gridcell_area_weights,
+                                              returned=True)
+
+    #TODO: Build in try/except block for better error handling
```

Comment:
@jrknezha could you please propose how to include a try/except block here that better handles potential assertion errors?

Reply:
Per our discussion, this would be better to try (asserts) then except (assertion error) where you can provide a clear error message and error of what this assert is actually checking for. this allows clearer information to users/developers about what is failing and the code can still fail on the error. 

Example syntax from score-db: 
try:
        description = json.loads(body.get('description'))
except Exception as err:
        msg = 'Error loading \'description\', must be valid JSON - err: {err}'
        raise ValueError(msg) from err